src/com/oltpbenchmark/api/Procedure.java

```diff
@@ -128,7 +128,8 @@ public final PreparedStatement getPreparedStatementReturnKeys(Connection conn, S
             
             // HACK: If the target system is Postgres, wrap the PreparedStatement in a special
             //       one that fakes the getGeneratedKeys().
-            if (is != null && this.dbType == DatabaseType.POSTGRES) {
+            if (is != null && this.dbType == DatabaseType.POSTGRES ||
+                (this.dbType == DatabaseType.HANA)) {
```

Comment:
Is this necessary? the AutoIncrementPreparedStatement doesn't have special handling for HANA anyway.